core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1055,6 +1080,32 @@ public synchronized void afterMerge(OnGoingMerge merge) {
                     deactivateThrottling();
                 }
             }
+            if (indexWriter.hasPendingMerges() == false && System.nanoTime() - lastWriteNanos >= engineConfig.getFlushMergesAfter().nanos()) {
```

Comment:
I guess because the `FlushMergesAfter` defaults to 5 minutes, and so does a sync'd flush because the index became inactive (`indices.memory.shard_inactive_time`), we shouldn't have a case where a merge finished on an inactive index but we never trigger a flush for it?  Maybe we should lower the default `FlushMergeAfter` a bit so the are not ==?